### PR TITLE
Switch message transport from websockets to stdin/stdout.

### PR DESCRIPTION
**Pull request by int19h**
(for remoting - corresponding RTVS PR coming shortly)

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -214,6 +183,7 @@ namespace rhost {
 }
 
 int main(int argc, char** argv) {
+    //MessageBox(0, 0, 0, 0);
```
Comment (AlexanderSher) [inline src/main.cpp]: ?

Comment (int19h) [inline src/main.cpp]: Debug helper :) I'll get rid of it.

File: src/transport.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        void send_message(const message& msg) {
+            assert(output);
+
+            log_message("<==", msg.id(), msg.request_id(), msg.name(), msg.json_text(), msg.blob());
+
+            auto& payload = msg.payload();
+            boost::endian::little_uint32_buf_t msg_size(static_cast<uint32_t>(payload.size()));
+
+            std::lock_guard<std::mutex> lock(output_lock);
+            if (fwrite(&msg_size, sizeof msg_size, 1, output) == 1) {
```
Comment (AlexanderSher) [inline src/transport.cpp]: What does `== 1` mean?

Comment (int19h) [inline src/transport.cpp]: `fwrite` and `fread` return the number of elements read or written. In this case we're writing 1 element (3rd argument here) of size `sizeof msg_size` (2nd argument); so if write succeeded, the return value will be 1, and if it failed or partially succeeded, it'll be 0.

**Comment by AlexanderSher** [comment]
:shipit:

**Comment by karthiknadig** [comment]
:shipit:

**Pull request merged**